e2e/testcases/resource_group_controller_test.go

```diff
@@ -169,33 +180,31 @@ func TestResourceGroupControllerInKptGroup(t *testing.T) {
 			},
 		},
 	}
-
 	nt.Must(nt.Watcher.WatchObject(kinds.ResourceGroup(), rgNN.Name, rgNN.Namespace,
 		testwatcher.WatchPredicates(
 			testpredicates.ResourceGroupStatusEquals(expectedStatus),
 		), testwatcher.WatchTimeout(time.Minute)))
 
-	if err := testresourcegroup.CreateOrUpdateResources(nt.KubeClient, resources, resourceID); err != nil {
-		nt.T.Fatal(err)
-	}
+	nt.T.Log("Simulate the Reconciler applying the managed resources")
+	nt.Must(testresourcegroup.CreateOrUpdateResources(nt.KubeClient, resources, resourceID))
 
-	expectedStatus.ObservedGeneration = 3
```

Comment:
Why was the observedGeneration expectation changed?

Reply:
It didn't. It was already being set above on line 148. This was a no-op.

Reply:
I see, that's fine but probably makes it a bit less clear as the reader needs to go reference the previous assignment